src/widgets/courses/ui/courses.test.tsx

```diff
@@ -0,0 +1,80 @@
+import { screen } from '@testing-library/react';
+import { Mock, describe, expect, it, vi } from 'vitest';
+
+import { Courses } from './courses';
+import { MOCKED_IMAGE_PATH } from '@/shared/__tests__/constants';
+import { renderWithRouter } from '@/shared/__tests__/utils';
+import { useDataByName } from '@/shared/hooks/use-data-by-name';
+
+const widgetTitle = 'All courses';
+const mockCourses = [
+  {
+    id: '1',
+    title: 'React course',
+    iconSrc: MOCKED_IMAGE_PATH,
+    startDate: '23 Oct, 2023',
+    language: ['ru', 'en'],
+    mode: 'online',
+    detailsUrl: 'https://rs.school/react/',
+    backgroundStyle: {
+      backgroundColor: '#EEF3FE',
+      accentColor: '#7356BF',
+    },
+  },
+  {
+    id: '2',
+    title: 'React course 2',
+    iconSrc: MOCKED_IMAGE_PATH,
+    startDate: '23 Oct, 2023',
+    language: ['ru', 'en'],
+    mode: 'online',
+    detailsUrl: 'https://rs.school/react/',
+    backgroundStyle: {
+      backgroundColor: '#EEF3FE',
+      accentColor: '#7356BF',
+    },
+  },
+];
+
+vi.mock('@/shared/hooks/use-data-by-name', () => {
+  return {
+    useDataByName: vi.fn(() => ({
+      data: mockCourses,
+      loading: false,
+      error: undefined,
+    })),
+  };
+});
+
+describe('Courses (other courses) component', () => {
+  it('renders widget without crashing and display correct content', () => {
+    renderWithRouter(<Courses />);
+    const widget = screen.getByTestId('all-courses');
+    const title = screen.getByTestId('widget-title');
+    const courseCard = screen.getAllByTestId('course-card');
+
+    expect(widget).toBeVisible();
+    expect(title).toBeVisible();
+    expect(title).toHaveTextContent(widgetTitle);
+    courseCard.forEach((card, index) => {
```

Comment:
Maybe we should also check how many cards are rendered?

Reply:
Added